doc/dev/test_proxy_migration_guide.md

```diff
@@ -420,6 +425,64 @@ container if it's not already running.
 
 For more details on proxy startup, please refer to the [proxy documentation][detailed_docs].
 
+### Use `pytest.mark.parametrize` with migrated tests
+
+Migrating tests to use basic `pytest` tools allows us to take advantage of helpful features such as
+[parametrization][parametrize]. Parametrization allows you to share test code by re-running the same test with varying
+inputs. For example, [`azure-keyvault-keys` tests][parametrize_example] are parametrized to run with multiple API
+versions and multiple Key Vault configurations.
+
+Because of how the `pytest.mark.parametrize` mechanism works, the `recorded_by_proxy(_async)` decorators aren't
+compatible without an additional decorator that handles the arguments we want to parametrize. The callable that
+`pytest.mark.parametrize` decorates needs to have positional parameters that match the arguments we're parametrizing;
+for example:
+
+```python
+import pytest
+from devtools_testutils import recorded_by_proxy
+
+test_values = [
+    ("first_value_a", "first_value_b"),
+    ("second_value_a", "second_value_b"),
+]
+
+# Works because `parametrize` decorates a method with positional `a` and `b` parameters
+@pytest.mark.parameterize("a, b", test_values)
+def test_function(a, b, **kwargs):
+    ...
+
+# Doesn't work; raises collection error
+# `recorded_by_proxy`'s wrapping function doesn't accept positional `a` and `b` parameters
+@pytest.mark.parameterize("a, b", test_values)
+@recorded_by_proxy
+def test_recorded_function(a, b, **kwargs):
+    ...
+```
+
+To parametrize recorded tests, we need a decorator between `pytest.mark.parametrize` and `recorded_by_proxy` that
+accepts the expected arguments. We can do this by declaring a class with a custom `__call__` method:
+
+```python
+class ArgumentPasser:
+    def __call__(self, fn):
+        # _wrapper accepts the `a` and `b` arguments we want to parametrize with
+        def _wrapper(test_class, a, b, **kwargs):
+            fn(test_class, a, b, **kwargs)
+        return _wrapper
+
+# Works because `ArgumentPasser.__call__`'s return value has the expected parameters
+@pytest.mark.parameterize("a, b", test_values)
```

Comment:
So functionally, we're placing this _first_ so that it ends up calling the rest in serial, yes?

EG,

parametrize() -> a passed forward
                      -> b passed forward
                      -> test_class passed forward

Reply:
It essentially multiplies the invocation of the script by the number of inputs, is basically what I'm saying.